### PR TITLE
PageResource layout API Swagger Annotations

### LINKED ISSUE #29143: Documentation for v1/page/layout
### Parent Issue

[29089](https://github.com/dotCMS/core/issues/29089)

### Task

Create Swagger documentation for the v1/page/layout and v1/page/{pageId}/layout paths in the [PageResource](https://github.com/dotCMS/core/blob/master/dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java) endpoint.

### Proposed Objective

Documentation

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

- [ ] Description of page layout endpoint
- [ ] Input method parameters
- [ ] Insure required fields are laid out

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by gavriella-dotcms**
### Proposed Changes
* Annotations made to the /{pageId}/layout Page endpoint
* Annotations made to the /layout Page endpoint

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

This PR fixes: #29143

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java [hunk 1]
```diff
@@ -403,9 +412,16 @@ private Response getPageRender(final HttpServletRequest originalRequest,
...
     @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
     @Path("/{pageId}/layout")
+    @Operation(operationId = "postPageLayout",
+                summary = "Links template and HTML",
+                description = "Takes a saved template and links it to an HTML page.\n\n" +
+                                "Any pages with a template already linked will update with the new link.\n\n" +
+                                "Otherwise a new template will be created without making any changes to previous templates.\n\n" +
+                                "Returns the updated page view for specified page.\n\n",
+                tags = {"Page"})
     public Response saveLayout(@Context final HttpServletRequest request,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:422]: I think you can replace the Response for a ResponseEntityView<Template> and also remove the exception handling; upper layer will take care about it

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java (hunk 1 above)
Comment (gavriella-dotcms) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:422]: Would this just involve removing the try/catch statements?

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/ResponseEntityPageView.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.dotcms.rest.api.v1.page;
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/ResponseEntityPageView.java:1]: @gavriella-dotcms what is this class for? I see some postman tests broken. Might it be related?

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java [hunk 3]
```diff
@@ -113,8 +122,7 @@
  * @version 4.2
  * @since Oct 6, 2017
  */
-@Path("/v1/page")
-@Tag(name = "Page")
+
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResource.java:125]: The @Path must not be removed. It breaks the endpoint url

**Review by nollymar — CHANGES_REQUESTED** [review]

**Review by nollymar — APPROVED** [review]
Checks pass now

**Pull request merged**